### PR TITLE
For #14923 - Removes auto size preference toggle

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAccessibilityRobot.kt [hunk 1]
```diff
@@ -49,13 +48,7 @@ class SettingsSubMenuAccessibilityRobot {
...
     }
 
-    fun verifyAutomaticFontSizingMenuItems() = assertAutomaticFontSizingMenuItems()
-
-    fun clickFontSizingSwitch() = toggleFontSizingSwitch()
-
-    fun verifyNewMenuItems() = assertNewMenuItems()
-
-    fun verifyNewMenuItemsAreGone() = assertNewMenuItemsAreGone()
+    fun verifyMenuItems() = assertNewMenuItems()
```
Comment (gabrielluong) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAccessibilityRobot.kt:51]: [optional][nit] Should we rename assertNewMenuItems() to assertMenuItems() as well?

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 2]
```diff
@@ -310,11 +310,6 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false
     )
 
-    var shouldUseAutoSize by booleanPreference(
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:313]: I am wondering if we can turn this into a getter (might not have to be in Settings or use the same name) and return the boolean value of `fontSizeFactor == 1F`. The way I think about this is that `fontSizeFactor == 1F` could be abstracted into a getter and with the right naming could help others reason what the setting means when looking at the other classes and not having the context of font size resizing.

**Review by gabrielluong — APPROVED** [review]
Changes look good. I would consider abstracting away `fontSizeFactor == 1F` into a getter to help provide a bit more context.

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt (hunk 2 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/utils/Settings.kt:319]: Good call out! Changed to reflect that and make it more usable

**Pull request merged**